src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -52,25 +45,20 @@ protected void addThreadExecutorToClient(Client client) {
    * manually.
    */
   public void wakeup() {
-    if(isRunning()) {
-      commonSelector.wakeup();
-    }
-  }
-
-  @Override
-  public void flagNewWrite(Client client) {
-    updateClientOps(client);
+    checkRunning();
+    commonSelector.wakeup();
   }
 
   @Override
-  public void flagNewRead(Client client) {
-    updateClientOps(client);
-  }
-  
-  private void updateClientOps(Client client) {
+  public void setClientOperations(final Client client) {
     ArgumentVerifier.assertNotNull(client, "Client");
-    if(isRunning() && clients.containsKey(client.getChannel())) {
-      if(client.canWrite() && client.canRead()) {
+    if(isRunning() && !clients.containsKey(client.getChannel()) && !client.isClosed() && client.getClientsSocketExecuter() == this) {
```

Comment:
Yeah, I have thought about removing this logic all together.  Technically there is no way a client or server could not be in the corresponding map unless it was closed.  This is kind of left over from when they could exist outside a socketexecuter.  I will look into removing it